extensions/wikia/AdEngine/js/PageFairDetection.js

```diff
@@ -0,0 +1,73 @@
+define('ext.wikia.adEngine.pageFairDetection', [
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.utils.scriptLoader',
+	'wikia.document',
+	'wikia.log',
+	'wikia.window'
+], function (adContext, scriptLoader, doc, log, window) {
+	'use strict';
+
+	var context = null,
+		logGroup = 'ext.wikia.adEngine.pageFairDetection';
+
+	function getContext() {
+		if (context === null) {
+			context = adContext.getContext();
+		}
+		return context;
+	}
+
+	function isMobile() {
+		var isMobileSkin = getContext().targeting.skin === 'mercury',
+			userAgentIsMobile = /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent);
+
+		return isMobileSkin || userAgentIsMobile;
+	}
+
+	function getWebsiteCode() {
+		var websiteKeys = {
+			mobile: 'FA8D90F8D9F54665',
+			desktop: 'FE3882548B7E471A'
+		};
+
+		return isMobile() ? websiteKeys.mobile : websiteKeys.desktop;
+	}
+
+	function dispatchDetectionEvent(eventName){
+		var event = document.createEvent('Event');
+		event.initEvent(eventName, true, false);
```

Comment:
Look what @fraszczakszymon did in his PR -> https://github.com/Wikia/app/pull/10901/files#diff-b9e82d7dddc23f114c5d710e787e5bdaR136 . I believe both of you want to achieve the same thing in a similar (not the same) way. How about unification of events creation?